lib/pos_blockchain.go

```diff
@@ -1375,26 +1375,25 @@ func (bc *Blockchain) tryApplyNewTip(blockNode *BlockNode, currentView uint64, l
 	}
 
 	// We need to perform a reorg here. For simplicity, we remove all uncommitted blocks and then re-add them.
-	committedTip, idx := bc.getCommittedTip()
-	if committedTip == nil || idx == -1 {
+	committedTip, committedTipIdx := bc.getCommittedTip()
+	if committedTip == nil || committedTipIdx == -1 {
 		// This is an edge case we'll never hit in practice since all the PoW blocks
 		// are committed.
 		return false, errors.New("tryApplyNewTip: No committed blocks found")
 	}
 	// Remove all uncommitted blocks. These are all blocks that come after the committedTip
-	// in the best chain.
-	// Delete all blocks from bc.bestChainMap that come after the highest committed block.
-	for ii := idx + 1; ii < len(bc.bestChain); ii++ {
-		delete(bc.bestChainMap, *bc.bestChain[ii].Hash)
+	// in the best chain. Remove all blocks from bc.bestChainMap that come after the highest
+	// committed block.
+	numBlocksToRemove := len(bc.bestChain) - committedTipIdx - 1
```

Comment:
Maybe it's safer for us to do a while !bc.BlockTip().IsCommitted() here instead of slicing and dicing? Would also allow us to get rid of the logic to get the block tip above and the annoying error we could hit there?

Reply:
Hmm, that is a great idea. It does simplify the logic above. As long as you feel that it's safe, I'm down for the simplification

Reply:
I do like it!